service/controller/clusterapi/v28/resource/machinedeployment/create.go

```diff
@@ -2,8 +2,49 @@ package machinedeployment
 
 import (
 	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/aws-operator/service/controller/clusterapi/v28/controllercontext"
+	"github.com/giantswarm/aws-operator/service/controller/clusterapi/v28/key"
 )
 
 func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCluster(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "finding machine deployment for cluster")
+
+		in := metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s=%s", key.LabelCluster, key.ClusterID(cr)),
+		}
+
+		out, err := r.cmaClient.ClusterV1alpha1().MachineDeployments(metav1.NamespaceAll).List(in)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		if len(out.Items) == 0 {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "did not find machine deployment for cluster")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling reconciliation")
+			return nil
+		}
+		if len(out.Items) != 1 {
+			return microerror.Maskf(executionFailedError, "expected 1 machine deployment got %d", len(out.Items))
+		}
```

Comment:
FMI: why we have only one machine deployment object? Shouldn't be there one per node?

Reply:
There should be one per node pool. For now we only try to make the cluster work with its current structure, but the new types. So we reconcile the `Cluster` and fetch the `MachineDeployment` using this temporary resource.

Reply:
Sorry I didn't realize the `Deployment` part.

What entity is supposed to create the `MachineDeployment` object? Is the goal to reconcile node pool based on that object?

Reply:
For now `api` creates the `MachineDeployment`. Later we have a controller for it and a node pool will be reconciled upon the `MachineDeployment`.